publish/git-setup.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+# Stores github credentials in a file
+# https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage
+
+set -euo pipefail
+
+program="$(basename "$0")"
+
+usage () {
+    cat <<USAGE
+Usage: ${program} -t tokenFile -u user -e email [-n name] [-c credentialsFile]
+
+Options:
+    -t|--tokenFile <tokenFile>             : A file containing the GitHub token (required)
+    -u|--user <user>                       : The GitHub user name (required)
+    -n|--name <name>                       : The full name used for commits (optional)
+    -e|--email <email>                     : The email used for commits (required)
```

Comment:
nit: swap this and the line above to keep the required items together and match the `Usage`